### PR TITLE
47 setup a formatter

### PR DESCRIPTION
**Pull request by rosepearson**
Move to using the Black formatter and fix all warnings:
 - [X] Add a GH Action for enforcing Black formatting for a PR
 - [x] Update the Spyder IDE to use the Black formatter and update files on save
 - [x] Update Wiki documentation to note Black formatting is expected

### CODE REVIEW

**Comment by rosepearson** [comment]
Finally got around to using Black to auto-format. I've decided to match its 88 line length preference for comments and docstrings - but thought I would do this as I go - so I don't go crazy. 

@jennan I thought I would ask for a quick review just in case there is any thing more you would recommend. I have:
* Set my IDE to use the Black formatter and auto-format on save
* Added the Black GH action for reformatting on creating a PR

File: .github/workflows/autoblack.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - name: Install Black
+        run: pip install black
+      - name: Run black --check .
+        run: black --check .
+      - name: If needed, commit black changes to the pull request
+        if: failure()
+        run: |
+          black .
+          git config --global user.name 'autoblack'
+          git config --global user.email 'cclauss@users.noreply.github.com'
```
Comment (jennan) [inline .github/workflows/autoblack.yml:26]: Maybe change the email to something different from the tool's author email.

**Comment by jennan** [comment]
Wooo, that looks very nice :). This will generate a large diff but, that's life (with a formater) 🤷. I didn't know autoblack, very cool. One remark, you may want to change the email used to commit for something else than the tool's author's email (`git config --global user.email 'cclauss@users.noreply.github.com'` in autoblack.yml).

**Review by jennan — APPROVED** [review]
Wooo, that looks very nice :). This will generate a large diff but, that's life (with a formater) 🤷. I didn't know autoblack, very cool. One remark, you may want to change the email used to commit for something else than the tool's author's email (git config --global user.email 'cclauss@users.noreply.github.com' in autoblack.yml).

**Comment by rosepearson** [comment]
> Wooo, that looks very nice :). This will generate a large diff but, that's life (with a formater) 🤷. I didn't know autoblack, very cool. One remark, you may want to change the email used to commit for something else than the tool's author's email (git config --global user.email '[cclauss@users.noreply.github.com](mailto:cclauss@users.noreply.github.com)' in autoblack.yml).


Yes - it's very cool isn't it! Do you think my email would be more appropriate as the main maintainer atm?

**Pull request merged**